spec/rubocop/cop/rspec/leading_subject_spec.rb

```diff
@@ -160,4 +160,113 @@ def helper_method
       end
     RUBY
   end
+
+  it 'flags subject below example group' do
+    expect_offense(<<-RUBY)
+      RSpec.describe User do
+        describe do
+          it { is_expected.to be_present }
+        end
+
+        subject { described_class.new }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Declare `subject` above any other `describe` declarations.
+      end
+    RUBY
+
+    expect_correction(<<-RUBY)
+      RSpec.describe User do
+        subject { described_class.new }
+        describe do
+          it { is_expected.to be_present }
+        end
+
+      end
+    RUBY
+  end
+
+  it 'flags subject below shared example group' do
+    expect_offense(<<-RUBY)
+      RSpec.describe User do
+        shared_examples_for 'used later' do
+          it { is_expected.to be_present }
+        end
+
+        subject { described_class.new }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Declare `subject` above any other `shared_examples_for` declarations.
+      end
+    RUBY
+
+    expect_correction(<<-RUBY)
+      RSpec.describe User do
+        subject { described_class.new }
+        shared_examples_for 'used later' do
+          it { is_expected.to be_present }
+        end
+
+      end
+    RUBY
+  end
+
+  it 'flags subject below include' do
+    expect_offense(<<-RUBY)
+      RSpec.describe User do
+        it_behaves_like 'a good citizen'
+
+        subject { described_class.new }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Declare `subject` above any other `it_behaves_like` declarations.
+      end
+    RUBY
+
+    expect_correction(<<-RUBY)
+      RSpec.describe User do
+        subject { described_class.new }
+        it_behaves_like 'a good citizen'
+
+      end
+    RUBY
+  end
+
+  it 'flags subject below include with a block' do
+    expect_offense(<<-RUBY)
+      RSpec.describe User do
+        it_behaves_like 'a good citizen' do
+          let(:used_in_shared_examples) { 'something' }
+        end
+
+        subject { described_class.new }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Declare `subject` above any other `it_behaves_like` declarations.
+      end
+    RUBY
+
+    expect_correction(<<-RUBY)
+      RSpec.describe User do
+        subject { described_class.new }
+        it_behaves_like 'a good citizen' do
+          let(:used_in_shared_examples) { 'something' }
+        end
+
+      end
+    RUBY
+  end
+
+  it 'flags subject below include with a blockpass' do
+    expect_offense(<<-RUBY)
+      RSpec.describe User do
+        block = ->{ }
+        it_behaves_like 'a good citizen', &block
+
+        subject { described_class.new }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Declare `subject` above any other `it_behaves_like` declarations.
+      end
+    RUBY
+
+    expect_correction(<<-RUBY)
+      RSpec.describe User do
+        block = ->{ }
```

Comment:
Ideally, the `block` would be below the subject too, right? This makes me thing, do we want to allow _anything_ above the `subject`?

Reply:
I knew this question is coming :D
Didn't want to make this example too different from its siblings by making it a `shared_examples_for '...' do |&block|`.

The style guide [is pretty clear about that](https://rspec.rubystyle.guide/#leading-subject):
> When `subject` is used, it should be the first declaration in the example group.

I'd rather not change the behaviour in a radical way here. I can think of e.g. helper methods or custom matchers, some may prefer to have them in a section above subject.